actors/paych/src/lib.rs

```diff
@@ -147,7 +147,7 @@ impl Actor {
                 sv.channel_addr
             )
         })?;
-        if pch_addr != svpch_id_addr {
+        if pch_addr.id().unwrap() != svpch_id_addr {
```

Comment:
nit: its a bit nicer to make svpch_id_addr an address to do the comparison because then there is no code path to panic

ALso svpch_id not svpch_id_addr